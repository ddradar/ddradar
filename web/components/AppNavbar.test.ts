import { publicUser } from '@ddradar/core/test/data'
import Oruga from '@oruga-ui/oruga-next'
import { bulmaConfig } from '@oruga-ui/theme-bulma'
import { RouterLinkStub } from '@vue/test-utils'
import { beforeEach, describe, expect, test, vi } from 'vitest'
import { createI18n } from 'vue-i18n'

import AppNavbar from '~~/components/AppNavbar.vue'
import useAuth from '~~/composables/useAuth'
import { mountAsync } from '~~/test/test-utils'

vi.mock('~~/composables/useAuth')

describe('components/AppNavbar.vue', () => {
  const stubs = { NuxtLink: RouterLinkStub }
  beforeEach(() => vi.mocked(useAuth).mockClear())

  describe.each(['ja', 'en'])('{ locale: %s } snapshot test', locale => {
    const i18n = createI18n({ legacy: false, locale })

    test('({ isLoggedin: false }) renders login button', async () => {
      // Arrange
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      vi.mocked(useAuth).mockResolvedValue({ isLoggedIn: ref(false) } as any)

      // Act
      const wrapper = await mountAsync(AppNavbar, {
        global: { plugins: [[Oruga, bulmaConfig], i18n], stubs },
      })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
    test('({ isLoggedin: true }) renders user name & logout button', async () => {
      // Arrange
      /* eslint-disable @typescript-eslint/no-explicit-any */
      vi.mocked(useAuth).mockResolvedValue({
        isLoggedIn: ref(true),
        id: ref(publicUser.id),
        name: ref(publicUser.name),
      } as any)
      /* eslint-enable @typescript-eslint/no-explicit-any */

      // Act
      const wrapper = await mountAsync(AppNavbar, {
        global: { plugins: [[Oruga, bulmaConfig], i18n], stubs },
      })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('events', () => {
    test('.navbar-burger@click sets "is-active" class', async () => {
      // Arrange
      const i18n = createI18n({ legacy: false, locale: 'ja' })
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      vi.mocked(useAuth).mockResolvedValue({ isLoggedIn: ref(false) } as any)

      // Act
      const wrapper = await mountAsync(AppNavbar, {
        global: { plugins: [[Oruga, bulmaConfig], i18n], stubs },
      })
      await wrapper.find('a.navbar-burger').trigger('click')

      // Assert
      expect(wrapper.find('a.navbar-burger').classes()).toContain('is-active')
      expect(wrapper.find('div.navbar-menu').classes()).toContain('is-active')
    })
  })
})
