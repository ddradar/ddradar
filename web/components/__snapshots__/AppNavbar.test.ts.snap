// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/AppNavbar.vue > { locale: en } snapshot test > ({ isLoggedin: false }) renders login button 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="navbar-item"
    >
      <img
        alt="DDRadar Logo"
        src="/assets/logo.svg"
      />
      <b>
        DDRadar
      </b>
    </a>
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger"
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
    >
      <a
        class="navbar-item"
      >
        Find User
      </a>
      <!--v-if-->
      
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Level(SP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Level(DP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Version
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            DDR 1st
          </a>
          <a
            class="navbar-item"
          >
            DDR 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 3rdMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 4thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 5thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX2
          </a>
          <a
            class="navbar-item"
          >
            DDR EXTREME
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA2
          </a>
          <a
            class="navbar-item"
          >
            DDR X
          </a>
          <a
            class="navbar-item"
          >
            DDR X2
          </a>
          <a
            class="navbar-item"
          >
            DDR X3 VS 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2013)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2014)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20 PLUS
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A3
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Courses
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            NONSTOP(A20)
          </a>
          <a
            class="navbar-item"
          >
            GRADE(A20)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            GRADE(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A3)
          </a>
          <a
            class="navbar-item"
          >
            GRADE(A3)
          </a>
          
        </div>
      </div>
      
    </div>
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Login
        </a>
        <div
          class="navbar-dropdown is-right"
        >
          <div
            class="buttons"
          >
            <button
              class="button is-info"
              type="button"
            >
              <span
                class="button-wrapper"
              >
                <span
                  class="icon"
                >
                  <i
                    class="mdi mdi-twitter mdi-24px"
                  />
                </span>
                <span
                  class=""
                >
                  
                  Login via Twitter
                  
                </span>
                <!--v-if-->
              </span>
            </button>
            <button
              class="button is-dark"
              type="button"
            >
              <span
                class="button-wrapper"
              >
                <span
                  class="icon"
                >
                  <i
                    class="mdi mdi-github mdi-24px"
                  />
                </span>
                <span
                  class=""
                >
                  
                  Login via GitHub
                  
                </span>
                <!--v-if-->
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`components/AppNavbar.vue > { locale: en } snapshot test > ({ isLoggedin: true }) renders user name & logout button 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="navbar-item"
    >
      <img
        alt="DDRadar Logo"
        src="/assets/logo.svg"
      />
      <b>
        DDRadar
      </b>
    </a>
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger"
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
    >
      <a
        class="navbar-item"
      >
        Find User
      </a>
      <a
        class="navbar-item"
      >
        Score List
      </a>
      
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Level(SP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Level(DP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Version
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            DDR 1st
          </a>
          <a
            class="navbar-item"
          >
            DDR 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 3rdMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 4thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 5thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX2
          </a>
          <a
            class="navbar-item"
          >
            DDR EXTREME
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA2
          </a>
          <a
            class="navbar-item"
          >
            DDR X
          </a>
          <a
            class="navbar-item"
          >
            DDR X2
          </a>
          <a
            class="navbar-item"
          >
            DDR X3 VS 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2013)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2014)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20 PLUS
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A3
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Courses
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            NONSTOP(A20)
          </a>
          <a
            class="navbar-item"
          >
            GRADE(A20)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            GRADE(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A3)
          </a>
          <a
            class="navbar-item"
          >
            GRADE(A3)
          </a>
          
        </div>
      </div>
      
    </div>
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Public User
        </a>
        <div
          class="navbar-dropdown is-right"
        >
          <div
            class="buttons"
          >
            <a
              class="button is-info"
            >
              <span
                class="button-wrapper"
              >
                <span
                  class="icon"
                >
                  <i
                    class="mdi mdi-account mdi-24px"
                  />
                </span>
                <span
                  class=""
                >
                  
                  MyPage
                  
                </span>
                <!--v-if-->
              </span>
            </a>
            <button
              class="button is-warning"
              type="button"
            >
              <span
                class="button-wrapper"
              >
                <!--v-if-->
                <span
                  class=""
                >
                  
                  Logout
                  
                </span>
                <!--v-if-->
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`components/AppNavbar.vue > { locale: ja } snapshot test > ({ isLoggedin: false }) renders login button 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="navbar-item"
    >
      <img
        alt="DDRadar Logo"
        src="/assets/logo.svg"
      />
      <b>
        DDRadar
      </b>
    </a>
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger"
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
    >
      <a
        class="navbar-item"
      >
        ユーザーを探す
      </a>
      <!--v-if-->
      
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          レベル(SP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          レベル(DP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          バージョン
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            DDR 1st
          </a>
          <a
            class="navbar-item"
          >
            DDR 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 3rdMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 4thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 5thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX2
          </a>
          <a
            class="navbar-item"
          >
            DDR EXTREME
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA2
          </a>
          <a
            class="navbar-item"
          >
            DDR X
          </a>
          <a
            class="navbar-item"
          >
            DDR X2
          </a>
          <a
            class="navbar-item"
          >
            DDR X3 VS 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2013)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2014)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20 PLUS
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A3
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          コース
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            NONSTOP(A20)
          </a>
          <a
            class="navbar-item"
          >
            段位認定(A20)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            段位認定(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A3)
          </a>
          <a
            class="navbar-item"
          >
            段位認定(A3)
          </a>
          
        </div>
      </div>
      
    </div>
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          ログイン
        </a>
        <div
          class="navbar-dropdown is-right"
        >
          <div
            class="buttons"
          >
            <button
              class="button is-info"
              type="button"
            >
              <span
                class="button-wrapper"
              >
                <span
                  class="icon"
                >
                  <i
                    class="mdi mdi-twitter mdi-24px"
                  />
                </span>
                <span
                  class=""
                >
                  
                  Twitterでログイン
                  
                </span>
                <!--v-if-->
              </span>
            </button>
            <button
              class="button is-dark"
              type="button"
            >
              <span
                class="button-wrapper"
              >
                <span
                  class="icon"
                >
                  <i
                    class="mdi mdi-github mdi-24px"
                  />
                </span>
                <span
                  class=""
                >
                  
                  GitHubでログイン
                  
                </span>
                <!--v-if-->
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`components/AppNavbar.vue > { locale: ja } snapshot test > ({ isLoggedin: true }) renders user name & logout button 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="navbar-item"
    >
      <img
        alt="DDRadar Logo"
        src="/assets/logo.svg"
      />
      <b>
        DDRadar
      </b>
    </a>
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger"
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
    >
      <a
        class="navbar-item"
      >
        ユーザーを探す
      </a>
      <a
        class="navbar-item"
      >
        スコア一覧
      </a>
      
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          レベル(SP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          レベル(DP)
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            LEVEL 1
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 2
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 3
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 4
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 5
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 6
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 7
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 8
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 9
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 10
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 11
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 12
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 13
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 14
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 15
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 16
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 17
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 18
          </a>
          <a
            class="navbar-item"
          >
            LEVEL 19
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          バージョン
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            DDR 1st
          </a>
          <a
            class="navbar-item"
          >
            DDR 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 3rdMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 4thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDR 5thMIX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX
          </a>
          <a
            class="navbar-item"
          >
            DDRMAX2
          </a>
          <a
            class="navbar-item"
          >
            DDR EXTREME
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA
          </a>
          <a
            class="navbar-item"
          >
            DDR SuperNOVA2
          </a>
          <a
            class="navbar-item"
          >
            DDR X
          </a>
          <a
            class="navbar-item"
          >
            DDR X2
          </a>
          <a
            class="navbar-item"
          >
            DDR X3 VS 2ndMIX
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2013)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution (2014)
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A20 PLUS
          </a>
          <a
            class="navbar-item"
          >
            DanceDanceRevolution A3
          </a>
          
        </div>
      </div>
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          コース
        </a>
        <div
          class="navbar-dropdown"
        >
          
          <a
            class="navbar-item"
          >
            NONSTOP(A20)
          </a>
          <a
            class="navbar-item"
          >
            段位認定(A20)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            段位認定(A20 PLUS)
          </a>
          <a
            class="navbar-item"
          >
            NONSTOP(A3)
          </a>
          <a
            class="navbar-item"
          >
            段位認定(A3)
          </a>
          
        </div>
      </div>
      
    </div>
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <a
          class="navbar-link"
        >
          Public User
        </a>
        <div
          class="navbar-dropdown is-right"
        >
          <div
            class="buttons"
          >
            <a
              class="button is-info"
            >
              <span
                class="button-wrapper"
              >
                <span
                  class="icon"
                >
                  <i
                    class="mdi mdi-account mdi-24px"
                  />
                </span>
                <span
                  class=""
                >
                  
                  マイページ
                  
                </span>
                <!--v-if-->
              </span>
            </a>
            <button
              class="button is-warning"
              type="button"
            >
              <span
                class="button-wrapper"
              >
                <!--v-if-->
                <span
                  class=""
                >
                  
                  ログアウト
                  
                </span>
                <!--v-if-->
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
